packages/vitest/src/node/execute.ts

```diff
@@ -156,17 +158,33 @@ export async function executeInViteNode(options: ExecuteOptions) {
         throw new Error('You can import mock only inside of a running test')
 
       const mock = (mockMap[suite] || {})[path] || resolveMockPath(path, root, nmName)
+
       if (mock === null) {
         const fsPath = getActualPath(path, nmName)
         const exports = mockObject(await request(fsPath, false))
         setCache(fsPath, { exports })
         return exports
       }
+
       if (typeof mock === 'function')
         return callFunctionMock(path, mock)
+
       return request(mock, true)
     }
 
+    const mockFile = async(path: string, nmName: string, factory?: () => any) => {
+      if (!suite)
+        throw new Error('You can import mock only inside of a running test')
+
+      if (!factory) {
+        const code = await importMock(path, nmName)
```

Comment:
I'm not sure if it should be imported on `mock`

I think it can be done only if needed, how it was before